### PR TITLE
School library associations

### PR DESCRIPTION
**Pull request by tylher**
- Created a new class `Classroom` which has `has-many` and `belong-to` relationship with the `Student` class
- Created `Rental` and `book` classes
- Used the `Rental` class to show the `many-to-many` relationship between `Person` and `Book` classes

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
```
Comment (Lordkaito) [inline book.rb:1]: - [x] Here, in your book class, you are missing a method to add rentals to a book. Something similar to this: 
```
def add_rental (person, date)
 Rental.new(date, self, person)
end
```
Please make sure to add this method.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
```
Comment (Lordkaito) [inline rental.rb:1]: - [ ] Here, you are not setting the associations properly. Right now, you need to set the "belongs-to" association between book and rentals, and the same association between person and rentals. So please make sure to add it.

File: person.rb [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
     @id = Random.rand(1...1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = rentals
     super()
   end
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (Lordkaito) [inline person.rb:13]: - [ ] In your person class, there should be a method for adding rentals, just as it should be in the book class as well. Remember that the rental class receives a person and a book, so you need to set them in the correspondent classes as well.

**Review by Lordkaito — CHANGES_REQUESTED** [review]
Hi @tylher, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (tylher) [reply book.rb:1]: Thank you for this 🙇‍♂️

File: person.rb (hunk 3 above)
Comment (tylher) [reply person.rb:13]: Okay @Lordkaito  thank you for the clarification

File: person.rb [hunk 4]
```diff
@@ -20,6 +20,11 @@ def can_use_services?
     true if is_of_age? || @parent_permission
   end
 
+  def add_rental()
+    @rental.push(rental)
+    rental.person = self
+  end
```
Comment (DeliceLydia) [inline person.rb]: - [ ] When creating a `rental` in this class you should send `book` and `date` as parameters and remember The `person` itself should be sent as a parameter to create the new rental achieving the 'has-many' association you can do:
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: book.rb [hunk 5]
```diff
@@ -7,4 +7,9 @@ def initialize(title, author, rentals)
     @author = author
     @rentals = rentals
   end
+
+  def add_rental(rental)
+    @rental.push(rental)
+    rental.book = self
+  end
```
Comment (DeliceLydia) [inline book.rb]: - [ ] Same as here you should send `person` and `date` as parameters and remember The `book` itself should be sent as a parameter to create the new rental achieving the 'has-many' association you can do:
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @tylher, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @tylher, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**